Client/Frontend/Settings/WebsiteDataManagementViewController.swift

```diff
@@ -0,0 +1,196 @@
+        tableView.separatorColor = UIColor.theme.tableView.separator
+        tableView.backgroundColor = UIColor.theme.tableView.headerBackground
+        let footer = ThemedTableSectionHeaderFooterView(frame: CGRect(width: tableView.bounds.width, height: SettingsUX.TableViewHeaderFooterHeight))
+        footer.showBottomBorder = false
+        tableView.tableFooterView = footer
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = UITableViewCell(style: .default, reuseIdentifier: nil)
+
+        let section = Section(rawValue: indexPath.section)!
+        switch section {
+        case .sites:
+            let site = siteRecords[indexPath.item]
+            cell.textLabel?.text = site.nameOfSite
+        case .showMore:
+            cell.textLabel?.text = "Show More"
+            cell.textLabel?.textColor = showMoreButtonEnabled ? UIColor.theme.general.highlightBlue : UIColor.gray
+            cell.accessibilityTraits = UIAccessibilityTraitButton
+            cell.accessibilityIdentifier = "ShowMoreWebsiteData"
+            showMoreButton = cell
+        case .button:
+            cell.textLabel?.text = Strings.SettingsClearAllWebsiteDataButton
+            cell.textLabel?.textAlignment = .center
+            cell.textLabel?.textColor = UIColor.theme.general.destructiveRed
+            cell.accessibilityTraits = UIAccessibilityTraitButton
+            cell.accessibilityIdentifier = "ClearAllWebsiteData"
+            clearButton = cell
+        }
+        return cell
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return NumberOfSections
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let section = Section(rawValue: section)!
+        switch section {
+        case .sites:
+            return siteRecords.count
+        case .showMore, .button:
+            return 1
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
+        let section = Section(rawValue: indexPath.section)!
+        switch section {
+        case .sites:
+            if self.tableView.isEditing {
+                return true
+            }
+        case .showMore:
+            if showMoreButtonEnabled {
+                return true
+            }
+        case .button:
+            return true
+        }
+        return false
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let section = Section(rawValue: indexPath.section)!
+        switch section {
+        case .sites:
+            break
+        case .showMore:
+            getAllWebsiteData()
+        case .button:
+            func clearwebsitedata(_ action: UIAlertAction) {
```

Comment:
Move the method outside of the `didSelectRowAt` it'll make it cleaner to understand